django_project/realtime/tasks/ash.py

```diff
@@ -36,42 +36,65 @@ def check_processing_task():
             task_status__iexact='SUCCESS'):
         task_id = ash.task_id
         result = AsyncResult(id=task_id, app=realtime_app)
-        ash.task_status = result.state
-        ash.save()
+        if result.state == 'SUCCESS':
```

Comment:
@ismailsunni Can you review this?

Reply:
So, we only update the state if it's SUCCESS? Any reason behind this? Why don't we update when it's FAILURE?

Reply:
Yes, so I was afraid I'm missing something, that's why I asked for the review.
CMIIW, then it should be `if not result.state == 'PENDING', right?

Reply:
Well, I just checked again in the celery documentation, there are more state: http://docs.celeryproject.org/en/latest/reference/celery.states.html#all-states

Perhaps we can use comparison like `state('PROGRESS') > state('SUCCESS')`.

But for short term, I think not Pending is enough.